reddit/reddit.go

```
@@ -28,32 +26,26 @@ type PostMetaData struct {
 }
 
 // GetTopPosts return the 20 top posts of /r/all
-func GetTopPosts() []PostMetaData {
+func GetTopPosts() ([]PostMetaData, error) {
```

It's often better to return an error instead of using `log.Fatal` inside a package so that you have more control over your error handling. With `log.Fatal`, it will call `os.Exit`, which ends your program. You may instead want to have an error page that you render.